packages/reason-relay/reason-relay-ppx/library/Subscription.re

```diff
@@ -0,0 +1,124 @@
+open Ppxlib;
+open Util;
+
+/**
+ * Check out the comments for makeFragment, this does the same thing but for subscriptions.
+ */
+let make = (~loc, ~moduleName) => {
+  let typeFromGeneratedModule = makeTypeAccessor(~loc, ~moduleName);
+  let valFromGeneratedModule = makeExprAccessor(~loc, ~moduleName);
+  let moduleIdentFromGeneratedModule = makeModuleIdent(~loc, ~moduleName);
+
+  Ast_helper.Mod.mk(
+    Pmod_structure([
+      [%stri include [%m moduleIdentFromGeneratedModule(["Utils"])]],
+      [%stri module Types = [%m moduleIdentFromGeneratedModule(["Types"])]],
+      [%stri
+        module Internal = {
+          type updaterFn =
+            (
+              ReasonRelay.RecordSourceSelectorProxy.t,
+              [%t typeFromGeneratedModule(["Types", "response"])]
+            ) =>
+            unit;
+
+          [@bs.deriving abstract]
```

Comment:
maybe `@bs.obj` could be used instead of `@bs.deriving abstract`, it only generates the maker function and not the accessors and if needed you can write the signature of the maker function which is not possible AFAIK with bs.deriving.

Reply:
I'll look into that, thank you!